TESTS/mbed_drivers/ticker/main.cpp

```diff
@@ -14,58 +14,95 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
-
-/*
- * Tests is to measure the accuracy of Ticker over a period of time
- *
- *
- * 1) DUT would start to update callback_trigger_count every milli sec, in 2x callback we use 2 tickers
- *    to update the count alternatively.
- * 2) Host would query what is current count base_time, Device responds by the callback_trigger_count
- * 3) Host after waiting for measurement stretch. It will query for device time again final_time.
- * 4) Host computes the drift considering base_time, final_time, transport delay and measurement stretch
- * 5) Finally host send the results back to device pass/fail based on tolerance.
- * 6) More details on tests can be found in timing_drift_auto.py
- */
-
 #include "mbed.h"
 #include "greentea-client/test_env.h"
 #include "utest/utest.h"
 #include "unity/unity.h"
 
-using namespace utest::v1;
+
+using utest::v1::Case;
 
 #define ONE_MILLI_SEC 1000
+#define TICKER_COUNT 16
+#define MULTI_TICKER_TIME_MS 100
 volatile uint32_t callback_trigger_count = 0;
 static const int test_timeout = 240;
 static const int total_ticks = 10;
 
+extern uint32_t SystemCoreClock; /* CPU clock */
```

Comment:
Is there a portable way to access to that declaration from an header ?

Reply:
on most platforms `osKernelGetSysTimerFreq` works
but look https://github.com/ARMmbed/mbed-os/issues/4986

Reply:
> Is there a portable way to access to that declaration from an header ?
  
I wish there was as this is part of cmsis, they declare it there as extern symbol in the cmsis startup files

Reply:
I noticed lately that there was some changes and now `osKernelGetSysTimerFreq` return `SystemCoreClock` for all platforms. It uses `osRtxSysTimerGetFreq` and there is no platform overriding it so below default implementation is used for all targets
```
...
osRtxInfo.kernel.sys_freq = SystemCoreClock;
...
__WEAK uint32_t osRtxSysTimerGetFreq (void) {
  return osRtxInfo.kernel.sys_freq;
}
```

So now we coul use `osKernelGetSysTimerFreq` instead `SystemCoreClock`